### PR TITLE
Make content outline visible and in expanded mode by default

### LINKED ISSUE #88800: Make content outline visible and in expanded view by default

### PR DESCRIPTION
**Pull request by harisrozajac**
**What is this feature?**

It makes the content outline visible and in expanded mode by default. 

**Why do we need this feature?**

This will increase awareness of the content outline feature, which could prove helpful with new functionalities like log pinning and filtering. However, it's a valid concern that users will not want the content outline visible and expanded by default, especially if they are running queries that produce one or two panels. 

To mitigate this, we've decided to persist visibility and expanded/collapsed modes in local storage.

Closes: #88800 

Please check that:
- [ ] It works as expected from a user's perspective.

### CODE REVIEW

**Comment by gelicia** [comment]
@thanos-karachalios How do we feel about this being configurable and defaulted to expanded? This way, if a customer does end up having an issue with it, we can change it from their config. I'm thinking 3 settings: "expanded", "collapsed", "hidden", with the default of "expanded"

**Comment by ifrost** [comment]
@thanos-karachalios / @diegoadams should we make it more responsive? I know that 75% of users use at least 1512 wide screens, but I'm concerned about the remaining 25% and use cases when users put two Explore tabs side-by-side.

Maybe we could at least remeber last section in local storage?

**Comment by diegoadams** [comment]
> Maybe we could at least remeber last section in local storage?

Storing the last state of the outline in local storage is a good idea. We can start with that and see how it goes. If we get feedback that it’s still interfering with users workflows, then we can make adjustments.

**Comment by harisrozajac** [comment]
> > Maybe we could at least remeber last section in local storage?
> 
> Storing the last state of the outline in local storage is a good idea. We can start with that and see how it goes. If we get feedback that it’s still interfering with users workflows, then we can make adjustments.

@diegoadams, @ifrost addressed this in https://github.com/grafana/grafana/pull/90283/commits/a24fcf0759aa75bcaf4cbcdd229b3c6b53b9b80d

File: public/app/features/explore/Explore.test.tsx [hunk 1]
```diff
@@ -225,4 +233,14 @@ describe('Explore', () => {
       expect(dataSourcePicker).toBeInTheDocument();
     });
   });
+
+  describe('Content Outline', () => {
+    it('should retrieve the last visible state from local storage', async () => {
+      const getBoolMock = jest.spyOn(store, 'getBool').mockReturnValue(false);
```
Comment (harisrozajac) [inline public/app/features/explore/Explore.test.tsx:239]: Not sure if this is the best option in terms of clarity. Yes, we invoke getBool in content outline and getBool is just a a wrapper around window.local.storage, but maybe we can spy on the local storage directly and look up the key. Or maybe do both?

File: public/app/features/explore/ContentOutline/ContentOutline.tsx [hunk 2]
```diff
@@ -57,12 +63,7 @@ export function ContentOutline({ scroller, panelId }: { scroller: HTMLElement |
     }, {});
   });
 
-  const scrollIntoView = (
-    ref: HTMLElement | null,
-    itemPanelId: string,
-    itemType: ITEM_TYPES | undefined,
-    customOffsetTop = 0
-  ) => {
+  const scrollIntoView = (ref: HTMLElement | null, customOffsetTop = 0) => {
```
Comment (harisrozajac) [inline public/app/features/explore/ContentOutline/ContentOutline.tsx:66]: Realized that two args here are not being used anymore

File: public/app/features/explore/Explore.test.tsx (hunk 1 above)
Comment (ifrost) [reply public/app/features/explore/Explore.test.tsx:239]: I think it's also fine this way. Sometimes we do it directly with local storage which is also fine, but you also need to remember to reset local storage state after the test.

**Review by ifrost — APPROVED** [review]

**Pull request merged**